imgui/core.pyx

```diff
@@ -1365,6 +1365,196 @@ def is_window_collapsed():
     return cimgui.IsWindowCollapsed()
 
 
+def begin_main_menu_bar():
+    """Begin a main menu bar.
+
+    .. visual-example::
+        :auto_layout:
+
+        if imgui.begin_main_menu_bar():
+            imgui.end_main_menu_bar()
+
+    Args:
+
+    Returns:
+        bool: True if main menu bar is created, False otherwise.
+
+    .. wraps::
+         bool BeginMainMenuBar()
+    )
+    """
+    cdef cimgui.bool opened = True
+    opened = cimgui.BeginMainMenuBar()
+    return opened
+
+
+def end_main_menu_bar():
+    """End a main menu bar.
+
+    .. visual-example::
+        :auto_layout:
+
+        if imgui.begin_main_menu_bar():
+            imgui.end_main_menu_bar()
+
+    Args:
+
+    Returns:
+
+    .. wraps::
+         bool EndMainMenuBar()
+    )
+    """
+    cimgui.EndMainMenuBar()
+
+
+def begin_menu_bar():
+    """Create a menu bar.
+
+    .. visual-example::
+        :auto_layout:
+
+        if imgui.begin_menu_bar():
+            imgui.end_menu_bar()
+
+    Args:
+
+    Returns:
+        bool: True if menu bar is created, False otherwise.
+
+    .. wraps::
+        bool BeginMenuBar()
+    )
+    """
+    return cimgui.BeginMenuBar()
+
+
+def end_menu_bar():
+    """End a menu bar item.
+
+    .. visual-example::
+        :auto_layout:
+
+        if imgui.begin_menu_bar():
+            imgui.end_menu_bar()
+
+    Args:
+
+    Returns:
+
+    .. wraps::
+         void EndMenuBar()
+    )
+    """
+    cimgui.EndMenuBar()
+
+
+def begin_menu(char* label, enabled=False):
+    """Create a menu.
+
+    .. visual-example::
+        :auto_layout:
+
+        if imgui.begin_menu("File"):
+            imgui.end_menu()
+
+    Args:
+        label (char *): label of the menu.
+        enabled (bool): define if menu is enabled or disabled.
+
+    Returns:
+        bool: If the menu was created successfully.
+
+    .. wraps::
+        bool BeginMenu(const char* label, bool enabled)
+    )
+    """
+    return cimgui.BeginMenu(label, enabled)
+
+
+def end_menu():
+    """End a menu.
+
+    .. visual-example::
+        :auto_layout:
+
+        if imgui.begin_menu():
+            imgui.end_menu()
+
+    Args:
+
+    Returns:
+
+    .. wraps::
+         void EndMenu()
+    )
+    """
+    cimgui.EndMenu()
+
+
+def menu_item(char* name, char* shortcut='', selected=False, enabled=True):
```

Comment:
Note that this is the design problem that bothered me for a while and is also reason why my development of this project slowed down. I have created a design document (see: #8) to organise my thoughts on this issue. You can comment and add some suggestions/opinions if you are interested.